datafusion/sqllogictest/test_files/sort_merge_join.slt

```diff
@@ -263,5 +263,139 @@ DROP TABLE t1;
 statement ok
 DROP TABLE t2;
 
+
```

Comment:
To verify that these tests cover the code changes, I ran them locally without the code changes in this PR and they failed as expected 👍 


```
Running "sort_merge_join.slt"
thread 'tokio-runtime-worker' panicked at datafusion/physical-plan/src/joins/sort_merge_join.rs:1356:22:
index out of bounds: the len is 0 but the index is 1
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
External error: task 17 panicked
Error: Execution("1 failures")
error: test failed, to rerun pass `-p datafusion-sqllogictest --test sqllogictests`

Caused by:
  process didn't exit successfully: `/Users/andrewlamb/Software/datafusion/target/debug/deps/sqllogictests-ce3a36cfeab74789 sort_merge` (exit status: 1)
```